packages/adblocker/src/engine/engine.ts

```diff
@@ -239,9 +238,26 @@ export default class FilterEngine extends EventEmitter<EngineEventHandlers> {
     return engine as InstanceType<T>;
   }
 
+  /**
+   * Merges compatible engines into one.
+   *
+   * This action references objects from the source engines,including
+   * network filters, cosmetic filters, preprocessors, metadata, and lists.
+   * These objects are not deep-copied, so modifying them directly can have
+   * unintended side effects.
+   * However, resources are deep-copied from the first engine.
+   *
+   * Optionally, you can specify a second parameter to skip merging specific resources.
+   * If resource merging is skipped, the resulting engine will be assigned empty resources.
+   */
   public static merge<T extends typeof FilterEngine>(
     this: T,
     engines: InstanceType<T>[],
+    {
+      skipResources = false,
+    }: {
+      skipResources?: boolean;
+    } = {},
```

Comment:
In which situation would we want to use `skipResources`?

Reply:
At Ghostery there are multiple reasons to skip:

1. When optional engine is enabled, it may have outdated resources
2. Some engines like for custom-filters have no resources at all